descriptive-link-5effbb.md

```diff
@@ -0,0 +1,179 @@
+---
+id: 5effbb
+name: Link text is descriptive
+rule_type: atomic
+
+description: | 
+ This rule checks that all links describe their purpose
+
+accessibility_requirements: 
+  wcag20:2.4.4: # Link Purpose (In Context) 
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+
+input_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Carlos Duarte
+- Marie Trudelle 
+- Ramit Garg
+---
+
+## Applicability
+
+The rule applies to any HTML element with the [semantic role](#semantic-role) of `link` that has an [accessible-name](#accessible-name) that is not the empty string ("").
+
+## Expectation
+
+Both the [text content](#text-content), if [visible](#visible), and the [accessible-name](#accessible-name) of the target element, together with its [visible](#visible) [context](#context) describe the purpose of the link.
```

Comment:
I think WCAG only asks us to have a good link text. Now I am uncertain what "link text" means (it has no definition) but we always interpreted it as the accessible-name of the link.

I would pass <a href="#' aria-label="read more about act rules">read more</a> while this rule would fail that scenario because 'read more' would not describe the purpose of the link on it's own.

Reply:
I think this mirrors the discussion about descriptive headings in https://github.com/act-rules/act-rules.github.io/pull/444.
That rule is kind of stranded, because we can't really agree on what we should be testing (visible/accessible name) and what the context is (visible/included in the accessibility tree).

Reply:
If we decide to have both visible name and accessible name as part of the expectation, I would prefer to have it at least split out in two expectations, which will allow for talking about which expectation failed/passed if it's not the same for both conditions.

Reply:
I've changed from 2 expectations to one expectation following your previous [comment](https://github.com/act-rules/act-rules.github.io/pull/555#discussion_r292893335) :-) 
But I'm happy to go back to the 2 expectations solution

Reply:
After looking at the proposed definition of "programmatically determined link context" I believe we can keep this with a single expectation, and a much simpler one